corehq/apps/integration/kyc/services.py

```diff
@@ -1,36 +1,174 @@
     pass
 
 
+def verify_users(user_objs, config):
+    # TODO: An endpoint to verify a group of users does not seem to be
+    #       available using Chenosis. If we have to do this with
+    #       multiple calls, consider using Celery gevent workers.
+    results = []
+    for user_obj in user_objs:
+        is_verified = verify_user(user_obj, config)
+        save_result(user_obj, config, is_verified)
+        results.append(is_verified)
+    return results
+
+
+def verify_user(user_obj, config):
+    """
+    Verify a user using the Chenosis MTN KYC API.
+
+    Returns True if all user data is accurate above its required
+    threshold, otherwise False.
+    """
+
+    # Documentation: https://apiportal.chenosis.io/en/apis/kyc-verify#endpoints
+    #
+    # Example request:
+    #
+    #     POST https://dev.api.chenosis.io/kycVerify/v1/customers/{customerId}
+    #     {
+    #         "firstName": "John",
+    #         "lastName": "Doe",
+    #         "phoneNumber": "27650600900",
+    #         "emailAddress": "john.doe@email.com",
+    #         "nationalIdNumber": "123456789",
+    #         "streetAddress": "1st Park Avenue, Mzansi, Johannesburg",
+    #         "city": "Johannesburg",
+    #         "postCode": "20200",
+    #         "country": "South Africa"
+    #     }
+    #
+    # Example 200 response:
+    #
+    #     {
+    #         "statusCode": "0000",
+    #         "statusMessage": "Success.",
+    #         "customerId": "27650600900",
+    #         "transactionId": "232TXYZ-212",
+    #         "data": {
+    #             "firstName": 100,
+    #             "lastName": 100,
+    #             "phoneNumber": 100,
+    #             "emailAddress": 100,
+    #             "nationalIdNumber": 100,
+    #             "streetAddress": 100,
+    #             "city": 100,
+    #             "postCode": 100,
+    #             "country": 100
+    #         }
+    #     }
+
+    # TODO: Determine what thresholds we want
+    required_thresholds = {
+        'firstName': 100,
+        'lastName': 100,
+        'phoneNumber': 100,
+        'emailAddress': 100,
+        'nationalIdNumber': 100,
+        'streetAddress': 80,  # Allow streetAddress to be less accurate
+        'city': 100,
+        'postCode': 100,
+        'country': 0,  # e.g "Ivory Coast" vs. "Republic of Côte d'Ivoire" vs. "CIV"?
+    }
+
+    user_data = get_user_data_for_api(user_obj, config)
+    _validate_schema('kycVerify/v1', user_data)
+    requests = config.get_connection_settings().get_requests()
+    response = requests.post(
+        f'/kycVerify/v1/customers/{user_data["phoneNumber"]}',
+        json=user_data,
+    )
+    response.raise_for_status()
+    field_scores = response.json().get('data', {})
+    return all(v >= required_thresholds[k] for k, v in field_scores.items())
+
+
+def save_result(user_obj, config, is_verified):
+    update = {
+        'kyc_provider': config.provider.value,  # TODO: Or config.provider.label?
```

Comment:
`config.provider.value` does not work as provider is the actual string value. 

```suggestion
        'kyc_provider': config.provider,  # TODO: Or config.provider.get_provider_display()?
```

Reply:
I am bit inclined towards storing the value here in case we want to make a conditional check on this later.